lib/ember_cli/shell.rb

```diff
@@ -37,6 +37,7 @@ def install
         exec "#{paths.bundler} install"
       end
 
+      exec "#{paths.ember} version || rm -rf #{paths.npm_deps} #{paths.bower_deps}"
```

Comment:
I like it! I think we'd also need to rename `Shell#exec` to `Shell#run!`, and introduce `Shell#run` to wrap `Runner#run`. If that sounds good to you, I'll proceed with all these changes.